pyulog/extract_message.py

```diff
@@ -0,0 +1,53 @@
+"""
+Extract values from a ULog file message to use in scripting
+"""
+
+import numpy as np
+from .core import ULog
+
+def extract_message(ulog_file_name: str, message: str, time_s: "int | None" = None, time_e: "int | None" = None,
+                     disable_str_exceptions=False) -> list[dict]:
+    """
+    Extract values from a ULog file
+
+    :param ulog_file_name: (str) The ULog filename to open and read
+    :param message: (str) A ULog message to return values from
+    :param time_s: (int) Offset time for conversion in seconds
+    :param time_e: (int) Limit until time for conversion in seconds
+
+    :return: (list[dict]) A list of each record from the ULog as key-value pairs
+    """
+
+    if not type(message) == str:
+        raise AttributeError("Must provide a message to pull from ULog file")
+
+    msg_filter = message.split(',') if message else None
+
+    ulog = ULog(ulog_file_name, msg_filter, disable_str_exceptions)
+    data = ulog.data_list
+
+    if not data:
+        raise AttributeError("Provided message is not in the ULog file")
```

Comment:
Can you use `ulog.get_dataset(message)` here?

Reply:
Yes I can. Looks like that method raises an IndexError if the message provided does not appear in the file, so I will raise an additional AttributeError in this extract_message module to be more descriptive about what caused the error.